modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectDefinitionDetails.tsx

```diff
@@ -42,71 +92,76 @@ export function RightSidebarObjectDefinitionDetails({
 					{sub(
 						Liferay.Language.get('x-details'),
 						getLocalizableLabel(
-							selectedNode?.data
-								.defaultLanguageId as Liferay.Language.Locale,
-							selectedNode?.data.label as LocalizedValue<string>,
-							selectedNode?.data.name
+							values.defaultLanguageId as Liferay.Language.Locale,
+							values?.label,
+							values?.name
 						)
 					)}
 				</span>
 			</div>
 			<div className="lfr-objects__model-builder-right-sidebar-definition-node-content">
 				<ObjectDataContainer
-					dbTableName={selectedNode?.data.dbTableName as string}
-					errors={{}}
-					handleChange={() => {}}
-					hasUpdateObjectDefinitionPermission={true}
-					isApproved={selectedNode?.data.status?.label === 'approved'}
-					setValues={() => {}}
-					values={selectedNode?.data as ObjectDefinition}
+					dbTableName=""
+					errors={errors}
+					handleChange={handleChange}
+					hasUpdateObjectDefinitionPermission={
+						values.actions
+							? 'update' in (values as ObjectDefinition).actions
+							: false
+					}
+					isApproved={values.status?.label === 'approved'}
+					setValues={setValues}
+					values={values as ObjectDefinition}
 				/>
 			</div>
 
 			<div className="lfr-objects__model-builder-right-sidebar-definition-node-content">
 				<EntryDisplayContainer
-					errors={{}}
-					nonRelationshipObjectFieldsInfo={[]}
-					objectFields={
-						selectedNode?.data.objectFields as ObjectField[]
+					errors={errors}
+					nonRelationshipObjectFieldsInfo={
+						nonRelationshipObjectFieldsInfo ?? []
 					}
-					setValues={() => {}}
-					values={selectedNode?.data as ObjectDefinition}
+					objectFields={(values?.objectFields as ObjectField[]) ?? []}
```

Comment:
Hey @igor-franca, since you are defining the Nullish coalescing operator (??) you don't need to type the (values?.objectFields as ObjectField[]), you can pass just the values?.objectFields and nullish will cover the rest.